src/waypoint_cmd.cpp

```diff
@@ -327,8 +328,15 @@ CommandCost CmdBuildBuoy(DoCommandFlag flags, TileIndex tile)
 		if (wp == nullptr) {
 			wp = new Waypoint(tile);
 		} else {
-			/* Move existing (recently deleted) buoy to the new location */
-			wp->xy = tile;
+			if (wp->xy != tile) {
+				/* Move existing (recently deleted) buoy to the new location. */
+				wp->xy = tile;
+				for (Ship *v : Ship::Iterate()) {
+					if (v->current_order.IsType(OT_GOTO_WAYPOINT)) {
+						v->SetDestTile(wp->xy);
+					}
+				}
```

Comment:
Please test this PR with ships going to different buoys, and then changing the location of one of the buoys.
As far as I am aware, this is not going to do what you intend it to do in that situation.

Reply:
oh, I see

Reply:
Solved. Also reduced identation while at it.